daq/network.py

```diff
@@ -267,22 +279,37 @@ def activate(self, native_gateway=None):
             # which caused the mininet host to have no IP
             self._used_ip_indices.add(self._get_host_ip_index(native_gateway.host))
 
-        self.tap_intf = self.ext_intf
-
-    def configure_remote_tap(self, remote):
+    def _configure_remote_tap(self, device):
         """Configure the tap for remote connection"""
+        if not device.session_endpoint:
+            return
+        remote = device.session_endpoint
         vxlan_config = self.config.get('switch_setup', {}).get('endpoint', {})
-        if 'ip' not in vxlan_config:
-            return False
-        remote_ip = remote.ip
-        vxlan_vni = vxlan_config.get('vni', self._DEFAULT_VXLAN_VNI)
-        dst_port = int(vxlan_config.get('port', self._DEFAULT_VXLAN_PORT))
-        vxlan_cmd = self._VXLAN_CMD_FMT % (self.ext_intf, vxlan_vni, remote_ip,
-                                           dst_port, dst_port, dst_port)
+        vxlan_port = self.topology.VXLAN_SEC_TRUNK_PORT + 1
+        while vxlan_port in self._vxlan_port_sets:
+            vxlan_port += 1
```

Comment:
is there an upper limit here?  What prevents this from getting to big?

Reply:
Theoretically there is no limit. When a device finishes testing, the port will be freed up for use by another device. Since the vxlan tunnel is maintained even if the device is queued for testing, a port needs to be allocated immediately.